src/cpp/rtps/builtin/discovery/participant/PDPServer2.hpp

```diff
@@ -124,12 +126,18 @@ class PDPServer2 : public aux::PDP
         mp_sync->restart_timer();
     }
 
+    //! Discovery database
+    fastdds::rtps::DiscoveryDataBase discovery_db;
+
 private:
 
     /**
      * TimedEvent for server synchronization
      */
     DServerEvent2* mp_sync;
+
+    //! PDP writer filter
+    IReaderDataFilter* pdp_filter_;
```

Comment:
again, Why keep the interface reference?

Reply:
done